libnmstate/nm/connection.py

```diff
@@ -137,36 +137,83 @@ def commit_profile(connection_profile, save_to_disk=True, nmdev=None):
     con_profile.commit(save_to_disk, nmdev)
 
 
+class ConnectionSetting(object):
+
+    def __init__(self, con_setting=None):
```

Comment:
```
if not con_setting:
    con_setting = nmclient.NM.SettingConnection.new()
    con_setting.props.uuid = str(uuid.uuid4())
    con_setting.props.autoconnect = True
    con_setting.props.autoconnect_slaves = (
￼	            nmclient.NM.SettingConnectionAutoconnectSlaves.YES)

# name, interface, type can be attributes to be set afterwards or we require either these values in init or con_setting

```

Reply:
This object represents a connection setting.
It may serve creating:
- A new one from scratch (`create`).
- Objectizing an existing setting, by getting the settings in the constructor. Consider this a copy constructor.
- Load/copy setting from the system/NM through a given profile.

Your suggestion supports partially options 1 (due to the missing args), fully option 2 and not option 3.
To support all, we will need to input it all, and then decide on an order or rules on how to read the input (to guess about what the caller intended).
If we are explicitly placing these options through different methods we make the api cleaner, and allow the methods/constructor focusing only on one thing, and not two.
It is also open to extensions (in case we will need to load the settings from other sources).

Reply:
Actually I believe that the code to get the profile from a device or by UUID should be moved to a device or client class. Since having the current objects is an improvement over not having them, I am also fine with doing a less strict review and revisit this later.

Reply:
I'm pretty sure the objects proposed here and in the next PR (for the device) will continue evolving.
This PR and refactoring in general is a never ending task and many times only after reorganizing stuff you see that things still need to move around.

I consider usage to dictate eventually what is better suited. Usage is usually what makes us refactor things in the first place.